src/actions/actions.ts

```diff
@@ -1652,6 +1652,114 @@ class MoveLineBegin extends BaseMovement {
 }
 
 @RegisterAction
+class MoveScreenLineBegin extends BaseMovement {
+  modes = [ModeName.Normal, ModeName.Visual, ModeName.VisualLine];
+  keys = ["g", "0"];
+  viewPosition = "wrappedLineStart";
+  /**
+   * This paramter will be consumed only when `to` is `lineUp` or `lineDown`.
+   * For other screen line movements, we are always operating on the same screen line.
+   * So we make its default value as 0.
+   */
+  noOfLines = 0;
+
+  public async execAction(position: Position, vimState: VimState): Promise<Position | IMovement> {
+    await vscode.commands.executeCommand("cursorMove", {
+      to: this.viewPosition,
+      inSelectionMode: vimState.currentMode !== ModeName.Normal,
+      noOfLines: this.noOfLines
+    });
+
+    if (vimState.currentMode === ModeName.Normal) {
+      return Position.FromVSCodePosition(vscode.window.activeTextEditor.selection.active);
+    } else {
+      /**
```

Comment:
I agree it's a bummer when users use this command quite often. Do we have any benchmark (or you have ideas about how) about commands to help measure how bad it is?

If it turns out to be unacceptablely slow, I'd like to push it hard from Code side.